File: tournaments/greatestCommonPrimeDivisor/greatestCommonPrimeDivisor.py
def greatestCommonPrimeDivisor(a, b):
    for i in range(min(a, b), 2, -1):
        pri = True
        for j in range(2, math.ceil(math.sqrt(i))):
            if i % j == 0:
                pri = False
                break
        if pri and a % i == 0 and b % i == 0:
            return i
    return -1

File: tournaments/greatestCommonPrimeDivisor/greatestCommonPrimeDivisor.js
function greatestCommonPrimeDivisor(a, b) {
    const isPrime = function(n) {
        for (let i = 2; n > i; ++i)
            if (!(n % i))
                return false;
        return true;
    };

    let result = -1;
    let min = Math.min(a, b);

    for (let i = 2; min >= i; ++i)
        if (isPrime(i) && !(a % i) && !(b % i) && result < i)
            result = i;

    return result;
}
